homeassistant/components/tplink/light.py

```diff
@@ -48,31 +48,25 @@
 LIGHT_STATE_ERROR_MSG = "err_msg"
 
 LIGHT_SYSINFO_MAC = "mac"
+LIGHT_SYSINFO_HOST = "host"
 LIGHT_SYSINFO_ALIAS = "alias"
 LIGHT_SYSINFO_MODEL = "model"
 LIGHT_SYSINFO_IS_DIMMABLE = "is_dimmable"
 LIGHT_SYSINFO_IS_VARIABLE_COLOR_TEMP = "is_variable_color_temp"
 LIGHT_SYSINFO_IS_COLOR = "is_color"
 
+MAX_ATTEMPTS = 20
+SLEEP_TIME = 2
+
 
 async def async_setup_entry(hass: HomeAssistantType, config_entry, async_add_entities):
     """Set up switches."""
-    await async_add_entities_retry(
-        hass, async_add_entities, hass.data[TPLINK_DOMAIN][CONF_LIGHT], add_entity
-    )
+    for device in hass.data[TPLINK_DOMAIN][CONF_LIGHT]:
+        await hass.async_add_executor_job(device.get_sysinfo)
+        async_add_entities([TPLinkSmartBulb(device)], update_before_add=True)
```

Comment:
You could rework this to add all entities at once, not sequentially.

Reply:
is there a good example of doing it all at once? I think that would drastically help this integration.